File: petcare-admin/src/api/department.ts
import request from '@/utils/request'
import type { ApiResult } from '@/types'

export interface Department {
  id: number
  name: string
  description: string
  icon: string
  sort: number
}

/** 获取所有启用科室列表（按sort排序） */
export function getDepartments() {
  return request.get<ApiResult<Department[]>>('/department/list')
}

File: petcare-server/petcare-system/src/main/java/com/petcare/system/entity/Department.java
package com.petcare.system.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@TableName("department")
public class Department {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String description;

    private String icon;

    private Integer sort;

    private Integer status;

    @TableLogic
    private Integer deleted;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}